config/v1/types_infrastructure.go

```diff
@@ -478,6 +478,20 @@ type AWSPlatformStatus struct {
 	// +kubebuilder:validation:MaxItems=25
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// userDNS indicates if the cluster is using a DNS solution configured
+	// by the user in place of Route53.
```

Comment:
What is the default value for this? Can this be modified at all? Is there any additional configuration needed now or maybe needed in the future that will warrant this being a struct with a type field rather than just a string?

Reply:
Default value is Disabled and cannot be modified once the cluster installation is complete. Hence the choice to add it to `Status` vs `Spec`. 
Yes, IMHO making this a type would make it future proof and I was toggling between this option (simple) vs the other (future proof).